package/src/components/AutoCompleteInput/AutoCompleteSuggestionItem.tsx

```diff
@@ -1,25 +1,137 @@
+        },
+      },
+    },
+  } = useTheme();
+  return (
+    <View style={[styles.container, mentionContainer]}>
+      <Avatar image={image} name={name} online={online} size={avatarSize} />
+      <View style={[styles.column, column]}>
+        <Text style={[styles.name, { color: black }, nameStyle]} testID='mentions-item-name'>
+          {name || id}
+        </Text>
+      </View>
+      <AtMentions pathFill={accent_blue} />
+    </View>
+  );
+};
+
+export const EmojiSuggestionItem = (item: Emoji) => {
+  const { native, name } = item;
+  const {
+    theme: {
+      colors: { black },
+      messageInput: {
+        suggestions: {
+          emoji: { container: emojiContainer, text },
+        },
+      },
+    },
+  } = useTheme();
+  return (
+    <View style={[styles.container, emojiContainer]}>
+      <Text style={[styles.text, { color: black }, text]} testID='emojis-item-unicode'>
+        {native}
+      </Text>
+      <Text style={[styles.text, { color: black }, text]} testID='emojis-item-name'>
+        {` ${name}`}
+      </Text>
+    </View>
+  );
+};
+
+export const CommandSuggestionItem = (item: CommandSuggestion) => {
+  const { args, name } = item;
+  const {
+    theme: {
+      colors: { black, grey },
+      messageInput: {
+        suggestions: {
+          command: { args: argsStyle, container: commandContainer, title },
+        },
+      },
+    },
+  } = useTheme();
+
+  return (
+    <View style={[styles.container, commandContainer]}>
+      {name ? <AutoCompleteSuggestionCommandIcon name={name} /> : null}
+      <Text style={[styles.title, { color: black }, title]} testID='commands-item-title'>
+        {(name || '').replace(/^\w/, (char) => char.toUpperCase())}
+      </Text>
+      <Text style={[styles.args, { color: grey }, argsStyle]} testID='commands-item-args'>
+        {`/${name} ${args}`}
+      </Text>
+    </View>
+  );
+};
+
+const renderSuggestionItem = (item: TextComposerSuggestion, triggerType?: string) => {
+  switch (triggerType) {
+    case 'mention':
+      return <MentionSuggestionItem {...(item as UserSuggestion)} />;
+    case 'emoji':
+      return <EmojiSuggestionItem {...(item as Emoji)} />;
+    case 'command':
+      return <CommandSuggestionItem {...(item as CommandSuggestion)} />;
+    default:
+      return null;
+  }
+};
+
+export const AutoCompleteSuggestionItem = ({
+  itemProps,
+  triggerType,
+}: AutoCompleteSuggestionItemProps) => {
+  const messageComposer = useMessageComposer();
+  const { textComposer } = messageComposer;
+
+  const {
+    theme: {
+      messageInput: {
+        suggestions: { item: itemStyle },
+      },
+    },
+  } = useTheme();
+
+  const handlePress = useCallback(async () => {
+    await textComposer.handleSelect(itemProps);
+  }, [itemProps, textComposer]);
+
+  return (
+    <Pressable
+      onPress={handlePress}
+      style={({ pressed }) => [{ opacity: pressed ? 0.2 : 1 }, itemStyle]}
+    >
+      {renderSuggestionItem(itemProps, triggerType)}
```

Comment:
This is again a function invocation that will create a new component in the DOM every time it wants to render something (which will be when the parent rerenders). Please convert `renderSuggestionItem` to an actual component that simply returns another and depends on actual props that would control how the DOM reconciles.

Reply:
Okay. I created the `renderSuggestionItem` outside the main component, so my assumption was this won't happen, but you might be correct here. Will keep this in mind.

Reply:
While `renderSuggestion` will keep stable, what it returns is essentially a new component. If you add it as a component, `React` reconciliation will kick in though.